test/alternative_dialog_spec.rb

```diff
@@ -45,47 +45,53 @@ def mock_selected_choice(*values)
   end
 
   describe "#auto_handler" do
-    it "calls Alternative#automatic_mode!" do
+    before do
       mock_ui_events(:auto)
+    end
+
+    it "calls Alternative#automatic_mode!" do
       expect(alternative).to receive(:automatic_mode!)
       dialog.run
     end
 
-    it "closes the dialog" do
-      mock_ui_events(:cancel)
-      expect(dialog).to receive(:finish_dialog).and_call_original
+    it "closes the dialog with true" do
+      expect(dialog).to receive(:finish_dialog).with(true).and_call_original
       dialog.run
     end
   end
 
   describe "#set_handler" do
-    it "calls Alternative#choose! with the path of the selected choice in the table" do
+    before do
       mock_ui_events(:set)
+    end
+
+    it "calls Alternative#choose! with the path of the selected choice in the table" do
       # Mock two values, first is used when open the dialog,
       # and the second is used when triggering set_handler.
       mock_selected_choice(alternative.value, "/usr/bin/vim")
       expect(alternative).to receive(:choose!).with("/usr/bin/vim")
       dialog.run
     end
 
-    it "closes the dialog" do
-      mock_ui_events(:cancel)
-      expect(dialog).to receive(:finish_dialog).and_call_original
+    it "closes the dialog with true" do
+      expect(dialog).to receive(:finish_dialog).with(true).and_call_original
       dialog.run
     end
   end
 
   describe "#cancel_handler" do
-    it "doesn't modify the alternative" do
+    before do
       mock_ui_events(:cancel)
+    end
+
+    it "doesn't modify the alternative" do
       expect(alternative).to_not receive(:choose!)
       expect(alternative).to_not receive(:automatic_mode!)
       dialog.run
     end
 
-    it "closes the dialog" do
-      mock_ui_events(:cancel)
-      expect(dialog).to receive(:finish_dialog).and_call_original
+    it "closes the dialog with false" do
+      expect(dialog).to receive(:finish_dialog).with(false).and_call_original
```

Comment:
does it really have to be mocked?
for me better test is to not mock this and instead check return value with 
```
  expect(dialog.run).to eq false
```

and same above for other checks of return value.